system_setup/server/controllers/configure.py

```diff
@@ -69,13 +66,88 @@ async def configure(self, *, context):
 
             self.app.update_state(ApplicationState.POST_RUNNING)
 
+            dryrun = self.app.opts.dry_run
+            variant = self.app.variant
+            config = WSLConfigHandler(dryrun)
+            if variant == "wsl_setup":
+                wsl_identity = self.model.identity.user
+                if dryrun:
+                    log.debug("mimicking creating user %s",
+                              wsl_identity.username)
+                else:
+                    run_command(["/usr/sbin/useradd", "-m", "-s", "/bin/bash",
```

Comment:
What happens when the command fails? Can we have something reported to the user os it silent here?

Reply:
it will report in info.log that the command exit with code 2

Reply:
I don’t think this is enough, as it won’t be user visible. From the user perspective: all is good, I haven’t seen any error.

We should surface the error to the user, I bet @mwhudson or @dbungert can guide you and we can grow an API to reflect to the user any errors both in the TUI and in ubuntu-desktop-installer (@jpnurmi, does this already exists?)